orchestrator/careplancontributor/ehr/kafka_client.go

```diff
@@ -0,0 +1,169 @@
+package ehr
+
+import (
+	"fmt"
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+	"github.com/labstack/gommon/log"
+	"os"
+	"strings"
+)
+
+// KafkaConfig holds the configuration settings for connecting to a Kafka broker.
+// It includes options to enable Kafka, specify the topic, endpoint, and connection string.
+//
+// Fields:
+//   - Enabled: A boolean indicating whether Kafka is enabled.
+//   - Topic: The Kafka topic to which messages will be sent.
+//   - Endpoint: The Kafka broker endpoint.
+//   - ConnectionString: The connection string used for authentication.
+type KafkaConfig struct {
+	Enabled  bool           `koanf:"enabled" default:"false"`
+	Topic    string         `koanf:"topic"`
+	Endpoint string         `koanf:"endpoint"`
+	Sasl     SaslConfig     `koanf:"sasl"`
+	Security SecurityConfig `koanf:"security"`
+}
+
+type SaslConfig struct {
+	Mechanism string `koanf:"mechanism" default:"PLAIN"`
+	Username  string `koanf:"username" default:"$ConnectionString"`
+	Password  string `koanf:"password"`
+}
+
+type SecurityConfig struct {
+	Protocol string `koanf:"protocol" default:"SASL_PLAINTEXT"`
+}
+
+// KafkaClient is an interface that defines the method for submitting messages.
+// Implementations of this interface should provide the logic for submitting
+// messages to a Kafka topic or an alternative storage.
+//
+// Methods:
+//   - SubmitMessage: Submits a message with the given key and value.
+//
+// Parameters:
+//   - key: The key of the message to be submitted.
+//   - value: The value of the message to be submitted.
+//
+// Returns:
+//   - error: An error if the message could not be submitted.
+type KafkaClient interface {
+	SubmitMessage(key string, value string) error
+}
+
+// KafkaClientImpl is an implementation of the KafkaClient interface.
+// It holds the Kafka topic and producer used to submit messages.
+type KafkaClientImpl struct {
+	topic    string
+	producer *kafka.Producer
+}
+
+type NoopClientImpl struct {
+}
+
+// NewClient creates a new KafkaClient based on the provided KafkaConfig.
+// If Kafka is enabled in the configuration, it initializes a Kafka producer
+// and sets up a delivery report handler for produced messages. If Kafka is
+// not enabled, it returns a NoopClientImpl which writes messages to a file.
+//
+// Parameters:
+//   - config: KafkaConfig containing the configuration for the Kafka client.
+//
+// Returns:
+//   - KafkaClient: An implementation of the KafkaClient interface.
+//   - error: An error if the Kafka producer could not be created.
+func NewClient(config KafkaConfig) (KafkaClient, error) {
+
+	var kafkaClient KafkaClient
+	if config.Enabled {
```

Comment:
feels a bit weird to not create a client if it's enabled. I.m.o. a better pattern is to have the caller decide to use the NoopClient (it's more explicit)? At least, that's how we do it for other services in ORCA (CPC, CPS, CPC's Task Engine)